src/python/pants/backend/go/subsystems/golang.py

```diff
@@ -102,6 +106,46 @@ def parse(v: str) -> tuple[int, int]:
 
         return parse(version) <= parse(self.version)
 
+    @property
+    def full_version(self) -> str:
+        return self.raw_metadata["GOVERSION"]
+
+    @property
+    def goos(self) -> str:
+        return self.raw_metadata["GOOS"]
+
+    @property
+    def goarch(self) -> str:
+        return self.raw_metadata["GOARCH"]
+
+    def arch_env(self) -> tuple[str, str] | None:
+        """Returns the name and setting of the GOARCH-specific architecture environment variable.
+
+        If the current architecture has no GOARCH-specific variable, returns empty key and value.
+        """
+
+        # See https://github.com/golang/go/blob/21998413ad82655fef1f31316db31e23e0684b21/src/cmd/go/internal/cfg/cfg.go#L261-L312
+        # for the original algorithm.
+        # TODO: It is not clear whether these would ever show up in `go env` output. The Go tool sources actually
+        # have some involved logic for determining the default. Maybe we need to vendor that logic as well?
+        # (although it looks like that code would need to be in Go)
+        if self.goarch == "arm":
+            return "GOARM", self.raw_metadata.get("GOARM", "")
+        elif self.goarch == "386":
+            return "GO386", self.raw_metadata.get("GO386", "")
+        elif self.goarch == "amd64":
+            return "GOAMD64", self.raw_metadata.get("GOAMD64", "")
+        elif self.goarch in ("mips", "mipsle"):
+            return "GOMIPS", self.raw_metadata.get("GOMIPS", "")
+        elif self.goarch in ("mips64", "mips64le"):
+            return "GOMIPS64", self.raw_metadata.get("GOMIPS64", "")
+        elif self.goarch in ("ppc64", "ppc64le"):
+            return "GOPPC64", self.raw_metadata.get("GOPPC64", "")
+        elif self.goarch == "wasm":
+            return "GOWASM", self.raw_metadata.get("GOWASM", "")
```

Comment:
FYI there's a pattern in Python to make this less clunky:

```python
env_var = {
   "arm": "GOARM",
    "386": "GO386",
    ...
}.get(self.goarch)
return (
   None
   if env_var is None
   else (env_var, self.raw_metadata.get(env_var, "")
)
```

Reply:
I may have this option "backwards." Apparently these variables are "micro-architectures" that allow selecting using non-default instruction sets in the machine language generated by the Go compiler. We currently don't support passing-through such options nor configuring them from a Pants option. These options will never be in `go env` output. 

Maybe this should just come out for now? Or be commented out until I can introduce the ability to set these micro-architectures in a subsequent PR?

Thoughts?

Reply:
Background: https://github.com/golang/go/issues/45453

Reply:
> Maybe this should just come out for now? 

Yeah, till this can be figured out properly. We try to not comment out stale code. 

I recommend a GitHub issue to track this

Reply:
https://github.com/pantsbuild/pants/issues/14663

Reply:
And removed.